enzy_htp/_interface/moe_interface.py

```diff
@@ -0,0 +1,109 @@
+"""Defines an MOEInterface class that serves as a bridge for enzy_htp to utilize MOE software. Uses the MOEConfig class
+found in enzy_htp/_config/moe_config.py. Supported operations include protonation.
+
+Author: Chris Jurich <chris.jurich@vanderbilt.edu>
+Date: 2023-04-01
+"""
+
+from pathlib import Path
+from typing import List, Dict
+
+from ..core.logger import _LOGGER
+from enzy_htp.core import file_system as fs
+from enzy_htp.core import env_manager as em
+
+from enzy_htp._config.moe_config import MOEConfig, default_moe_config
+
+#TODO(CJ): need to add tests for this section
+
+from .base_interface import BaseInterface
+
+
+class MOEInterface(BaseInterface):
+    """Class that provides a direct interface for enzy_htp to utilize MOE software. Supported operations 
+    include protonation. Users should ue this class as the only way to interact with any functionality 
+    in MOE.
+
+    Attributes:
+        config_	: The MOEConfig() class which provides settings for both running MOE and maintaining a compatible environment.
+        env_manager_ : The EnvironmentManager() class which ensure all required environment elements exist.
+        compatible_env_ : A bool() indicating if the current environment is compatible with the object itself.
+    """
+
+    def __init__(self, parent, config: MOEConfig = None) -> None:
+        """Simplistic constructor that optionally takes an MOEConfig object as its only argument.
+        Calls parent class.
+        """
+        super().__init__(parent, config, default_moe_config)
+
+    def valid_license(self) -> bool:
+        """MOE operates using a license based model. This function checks if a valid license is available in the system."""
+        pass
+
+    def _run_moebatch(self, input_svl: str) -> None:
+        """Helper function that runs the supplied scientific vector language (SVL) file as input with moebatch.
+        Method checks that the supplied svl file exists and has the correct file extension.
+
+        Args:
+            input_svl: A str() with the path of the .svl file.
+
+        Returns:
+            Nothing.
+        """
+
+        fs.check_file_exists(input_svl)
+
+        if Path(input_svl).suffix != '.svl':
+            _LOGGER.error(f"The supplied svl file '{input_svl}' does not have the extension .svl. Exiting...")
+            exit(1)
+
+        self.env_manager_.run_command(self.config_.MOE_BATCH, ["-script", str(input_svl)])
+
+    def protonate(self, molfile: str, outfile: str = None, pH: float = 7.0) -> str:
```

Comment:
Cool interface! We should also write an engine mimicing protonate_ligand_with_pybel() in preparation.protonate.protonate_ligand_with_moe() taking enzy_htp.Structure as input and make changes in place.

Reply:
This change is not necessary for this PR tho

Reply:
I think it would be better to do something along the lines of `preparation.protonate.protonate_ligand(fname,method="MOE")`.

Reply:
The engine option is provided in `protonate_stru(stru, ph, engine, ligand_engine)`. And it will call `preparation.protonate.protonate_ligand_with_moe()` where there's additional handling from and to `enzy_htp.Structure` in addition to just "molfile" based IO. (the handling will be different for different engines. e.g.: pybel needs to fix the H naming)